compiler-rt/lib/asan/asan_interceptors.cpp

```diff
@@ -352,8 +352,16 @@ static void ClearShadowMemoryForContextStack(uptr stack, uptr ssize) {
   PoisonShadow(bottom, ssize, 0);
 }
 
+// Since Solaris 10/SPARC, ucp->uc_stack.ss_sp refers to the stack base address
```

Comment:
Don't you need some kind of version check? Also, I suppose there's no need to intercept the old symbol to handle executables using old ABI, correct?

Reply:
LLVM only supports Solaris 11.4 for quite some time, it doesn't even build on 11.3 let alone 9.

Right: old executables could only be exposed to `libclang_rt.asan.so` via `LD_PRELOAD`, and I have a hard time believing anyone is still running Solaris 9 these days and suddendly develops a need for asan.  These days, you could only access the old `makecontext` if you declare it yourself: when including `<ucontext.h>`, you only get the v2 symbol on sparc.